src/system/SystemPacketBufferInternal.h

```diff
@@ -82,6 +82,20 @@
 #define CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_CUSTOM_POOL 0
 #endif
 
+/**
+ * CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_RAM_HEAP
+ *
+ * True if packet buffers are allocated from LwIP RAM Heap.
+ *
+ * Platform developer should implement and maintain their own customized function size_t lwip_mem_size(const void *p_mem).
+ *
+ */
+#if CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL && LWIP_PBUF_FROM_CUSTOM_RAM_HEAP
+#define CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_RAM_HEAP 1
```

Comment:
These two are mutually exclusive. How can you have `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL` and ` CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_RAM_HEAP` ? If it's from pool, it's not from heap. It seems like there is a need to update BUILD.gn to add more options for lwip memory config.

Reply:
Since PBUF_POOL is a type of enum in lwip `pbuf.h`, whatever `CHIP_SYSTEM_CONFIG_PACKETBUFFER_LWIP_PBUF_TYPE` defines, `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL` will be always `1` in the system built with lwip.

I guess two below options are added with `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL` protected
- `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_STANDARD_POOL` for lwip pbuf_pool
- `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_CUSTOM_POOL` for lwip cutom pool

Then, I add `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_RAM_HEAP` for third option: lwip ram heap.

So, `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_POOL` and `CHIP_SYSTEM_PACKETBUFFER_FROM_LWIP_RAM_HEAP` are not mutually exclusive.